pkg/game/elements/placed_tile.go

```diff
@@ -0,0 +1,72 @@
+package elements
+
+import (
+	"fmt"
+
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/tiles"
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/tiles/side"
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/tilesets"
+)
+
+type Position struct {
+	// int8 would be fine for base game (72 tiles) but let's be a bit more generous
+	x int16
+	y int16
+}
+
+func NewPosition(x int16, y int16) Position {
+	return Position{x, y}
+}
+
+func (pos Position) X() int16 {
+	return pos.x
+}
+
+func (pos Position) Y() int16 {
+	return pos.y
+}
+
+func (pos Position) MarshalText() ([]byte, error) {
+	return fmt.Appendf([]byte{}, "%v,%v", pos.x, pos.y), nil
+}
+
+func (pos *Position) UnmarshalText(text []byte) error {
+	_, err := fmt.Sscanf(string(text), "%v,%v", &pos.x, &pos.y)
+	return err
+}
+
+// https://wikicarpedia.com/car/Game_Figures
+type MeepleType uint8
+
+const (
+	NormalMeeple MeepleType = iota
+
+	MeepleTypeCount int = iota
+)
+
+type Meeple struct {
+	Player Player
+	Side   side.Side
+	Type   MeepleType
+}
+
+type LegalMove struct {
+	tiles.Tile
+	Pos Position
+}
+
+type PlacedTile struct {
```

Comment:
PlacedTile has a Meeple, while LegalMove doesn't which is what I thought was going to the network. If not then we should probably have another method for generating valid moves for a specific player as well.

Reply:
`LegalMove`s (coordinates + rotation) to put a tile on are player-agnostic but `PlacedTile`s (coordinates + rotation + whether meeple is placed + what meeple is placed) aren't since it depends on whether the player has a certain meeple.

If a list of valid `PlacedTile`s is the input to neural network then we need to have a way to generate these, no? It would be based on existing methods for generating legal moves but would have to instead return a bunch of `PlacedTile`s for each move to include information about all valid meeple placements as well.

Reply:
Well, it would be more accurate to say that the moves for the player would be a subset of all legal placements of the tile on the board *but* the current `LegalMove` API doesn't make that a possibility either since it doesn't actually contain a `Meeple`...

Yeah, this needs a small redesign...